teacherchannel/module.py

```diff
@@ -0,0 +1,248 @@
+
+    @commands.check(check.acl)
+    @teacherchannel_teacher.command(name="set", aliases=["add"])
+    async def teacherchannel_teacher_set(
+        self, ctx, channel: nextcord.TextChannel, teacher: nextcord.Member
+    ):
+        """<master or slave channel> <teacher member>
+        Mark user as a teacher of a channel. Permission sync will not apply to this member. Grants permissions to the teacher channel (slave) only."""
+        config = TeacherChannelDB.add_teacher(ctx.guild.id, channel.id, teacher.id)
+        if not config:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "Could not add teacher. Either the channel is not a teacher channel or this user has already been added.",
+                )
+            )
+            return
+        if channel.id != config.slave_id:
+            channel = ctx.guild.get_channel(config.slave_id)
+        await channel.set_permissions(teacher, read_messages=True, send_messages=True)
+        await ctx.reply(_(ctx, "Teacher added."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher {teacher.display_name} added to teacher channel {channel.name}.",
+        )
+
+    @commands.check(check.acl)
+    @teacherchannel_teacher.command(name="unset", aliases=["remove", "delete"])
+    async def teacherchannel_teacher_unset(
+        self, ctx, channel: nextcord.TextChannel, teacher: nextcord.Member
+    ):
+        """<master or slave channel> <teacher member>
+        Unmark user as a teacher of a channel. Permission overrides will be removed from the teacher channel (slave)."""
+        success = TeacherChannelDB.remove_teacher(ctx.guild.id, channel.id, teacher.id)
+        if not success:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "Could not remove teacher that was not there in the first place.",
+                )
+            )
+            return
+        config = TeacherChannelDB.get(ctx.guild.id, channel.id)
+        if channel.id != config.slave_id:
+            channel = ctx.guild.get_channel(config.slave_id)
+        await channel.set_permissions(teacher, overwrite=None)
+        await ctx.reply(_(ctx, "Teacher removed."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher {teacher.display_name} removed from teacher channel {channel.name}.",
+        )
+
+    @tasks.loop(seconds=10.0, count=1)
+    async def load_deltas(self):
+        guild_ids = [i.guild_id for i in TeacherChannelDB.get_guilds()]
+        for guild_id in guild_ids:
+            guild = self.bot.get_guild(guild_id)
+            channels_to_sync = TeacherChannelDB.get_all(guild_id)
+            for teacherchannel in channels_to_sync:
+                master = guild.get_channel(teacherchannel.master_id)
+                slave = guild.get_channel(teacherchannel.slave_id)
+                if not master:
```

Comment:
I see in this condition blok a lot of repeating code. I think there is some space for optimizing by for example not having separate `guild_log.warning(...)` calls, but using one with different texts as argument.